WordPress/WordPressTest/ReaderSiteServiceTests.swift

```diff
@@ -68,9 +64,137 @@ class ReaderSiteServiceTests: CoreDataTestCase {
+                    done()
+                }
+            )
+        }
+    }
+
+    func testFlagAsUnblockedSuccessPath() {
+        let service = makeService()
+        let siteID: NSNumber = 42
+
+        stub(condition: isPath("/rest/v1.1/me/block/sites/\(siteID)/delete")) { _ in
+            HTTPStubsResponse(jsonObject: ["success": 1], statusCode: 200, headers: nil)
+        }
+
+        waitUntil { done in
+            service.flagSite(
+                withID: siteID,
+                asBlocked: false,
+                success: {
+                    done()
+                },
+                failure: { error in
+                    // We call done in this failure scenario because we explicitly fail afterwards.
+                    // Without this call, the test would have two failures:
+                    // one because done was not called (timeout) and the explicit fail with error
+                    done()
+                    fail("Expected call to succeed. Failed with \(error?.localizedDescription ?? "'nil-error'")")
+                }
+            )
+        }
+    }
+
+    func testFlagAsUnblockedFailurePath() {
+        let service = makeService()
+        let siteID: NSNumber = 42
+
+        stub(condition: isPath("/rest/v1.1/me/block/sites/\(siteID)/delete")) { _ in
+            HTTPStubsResponse(jsonObject: ["success": 0], statusCode: 200, headers: nil)
+        }
+
+        waitUntil { done in
+            service.flagSite(
+                withID: siteID,
+                asBlocked: false,
+                success: {
+                    // We call done in this failure scenario because we explicitly fail afterwards.
+                    // Without this call, the test would have two failures:
+                    // one because done was not called (timeout) and the explicit fail with error
+                    done()
+                    fail("Expected call to fail, but succeeded")
+                },
+                failure: { error in
+                    expect((error as? NSError)?.domain) == ReaderSiteServiceRemoteErrorDomain
+                    expect((error as? NSError)?.code) == Int(ReaderSiteServiceRemoteError.sErviceRemoteUnsuccessfulBlockSite.rawValue)
+                    done()
+                }
+            )
+        }
+    }
+}
+
+extension ReaderSiteServiceTests {
+
+    func makeService(
+        username: String = "testuser",
+        authToken: String = "authtoken"
+    ) -> ReaderSiteService {
+        return makeService(username: username, authToken: authToken, contextManager: contextManager)
+    }
+
+    func makeService(
+        username: String,
+        authToken: String,
+        contextManager: ContextManager
+    ) -> ReaderSiteService {
+        let accountService = AccountService(coreDataStack: contextManager)
+        accountService.createOrUpdateAccount(withUsername: username, authToken: authToken)
+        return ReaderSiteService(coreDataStack: contextManager)
+    }
```

Comment:
This could have been a single method, to be fair. I just liked the idea of having one where we could inject the `ContextManager` value, even though we currently always use the one from the super class.